functests/gatekeeper/gatekeeper.go

```diff
@@ -0,0 +1,684 @@
+package gatekeeper
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	testClient "github.com/openshift-kni/cnf-features-deploy/functests/utils/client"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/namespaces"
+	"github.com/openshift-kni/cnf-features-deploy/functests/utils/pods"
+	corev1 "k8s.io/api/core/v1"
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	gkv1alpha "github.com/open-policy-agent/gatekeeper/apis/mutations/v1alpha1"
+)
+
+const (
+	// TestingNamespace is the namespace for resources in this test
+	TestingNamespace = "gatekeeper-testing"
+)
+
+var _ = Describe("gatekeeper", func() {
+	client := testClient.Client
+
+	AfterEach(func() {
+		err := deletePods(TestingNamespace, client)
+		Expect(err).NotTo(HaveOccurred())
+
+		namespacesUsed := []string{"mutation-included", "mutation-excluded", "gk-test-object"}
+		for _, namespace := range namespacesUsed {
+			if namespaces.Exists(namespace, client) {
+				err := client.Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+				namespaces.WaitForDeletion(client, namespace, time.Minute)
+			}
+		}
```

Comment:
If you care about test time, then maybe move the WaitForDeletion after all namespaces are deleted: 

```
		for _, namespace := range namespacesUsed {
			if namespaces.Exists(namespace, client) {
				err := client.Namespaces().Delete(context.Background(), namespace, metav1.DeleteOptions{})
				Expect(err).ToNot(HaveOccurred())
			}
		}

		for _, namespace := range namespacesUsed {
			if namespaces.Exists(namespace, client) {
				namespaces.WaitForDeletion(client, namespace, time.Minute)
			}
		}
```

Reply:
the namespaces i delete are used only on the last test, all other times we only delete the pods from the TestingNamespace.

Reply:
If time is not of essence we can skip this

Reply:
fixed